### ISSUE TITLE
timeouts when saltutil.refresh_modules/sync_all/refresh_pillar with multi-master

### ISSUE THREAD

**Issue by CaesarC** [saltstack/salt#19748]
```some things like``` #10900 #19708

```
#ping ok
[root@localhost ~]# salt \* test.ping
Executing job with jid 20150115093544191436
-------------------------------------------

id_1:
    True
id_1:
    True

```

```
#lastest version
[root@localhost ~]# salt \* test.versions_report
id_1:
               Salt: 2014.7.0
             Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
             Jinja2: unknown
           M2Crypto: 0.20.2
     msgpack-python: 0.1.13
       msgpack-pure: Not Installed
           pycrypto: 2.0.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.3.1
               RAET: Not Installed
                ZMQ: 4.0.4
               Mako: Not Installed
id_2:
               Salt: 2014.7.0
             Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
             Jinja2: unknown
           M2Crypto: 0.20.2
     msgpack-python: 0.1.13
       msgpack-pure: Not Installed
           pycrypto: 2.0.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.3.1
               RAET: Not Installed
                ZMQ: 4.0.4
               Mako: Not Installed  

```

```
#after sync_all,all minion will timeout
[root@localhost ~]# salt \* saltutil.sync_all -v
Executing job with jid 20150115095816033131
-------------------------------------------

id_1:
    ----------
    grains:
    modules:
    outputters:
    renderers:
    returners:
    states:
    utils:
id_1:
    ----------
    grains:
    modules:
    outputters:
    renderers:
    returners:
    states:
    utils:
```

```
#ping timeout
[root@localhost ~]# salt \* test.ping -v
Executing job with jid 20150115093544191436
-------------------------------------------

id_1:
    Minion did not return. [No response]
id_2:
    Minion did not return. [No response]

```

```
#multi-master

# Set the location of the salt master server. If the master server cannot be
# resolved, then the minion will fail to start.
master: 
  - 192.168.1.125 
  - 192.168.1.107

it also happend with saltutil.refresh_modules/refresh_pillar
```

**Comment by rallytime** [comment]
Ah, thanks for the report. We will definitely look into this.

**Comment by cachedout** [comment]
Fixed in #19790

**Comment by cachedout** [comment]
I am going to go ahead and close this because we now have a fix into the RC that has just been announced . If this has not been fixed to everybody's satisfaction, please let me know and we will be happy to-reopen it. Thanks!
